src/plugins/stake-plugins/generic/policyAdapters/TarotPoolAdaptor.ts

```diff
@@ -0,0 +1,491 @@
+
+      const bAmountAMin = tarot.decimalToBalance(amounts.bAmountAMin)
+      const bAmountBMin = tarot.decimalToBalance(amounts.bAmountBMin)
+
+      txs.push(
+        async () =>
+          await tarotRouterContract.populateTransaction.deleverage(poolContractAddress, collateralBalance, bAmountAMin, bAmountBMin, deadline, '0x', {
+            gasPrice,
+            gasLimit: '1000000',
+            customData: {
+              name: metadataName,
+              category: 'Transfer:Staking',
+              notes: `Deleverage ${metadataPoolAssetName} liquidity pool contract`
+            }
+          })
+      )
+
+      const allocations: QuoteAllocation[] = [
+        {
+          allocationType: 'unstake',
+          pluginId: policyConfig.stakeAssets[0].pluginId,
+          currencyCode: policyConfig.stakeAssets[0].currencyCode,
+          nativeAmount: bAmountAMin.toString()
+        },
+        {
+          allocationType: 'unstake',
+          pluginId: policyConfig.stakeAssets[1].pluginId,
+          currencyCode: policyConfig.stakeAssets[1].currencyCode,
+          nativeAmount: bAmountBMin.toString()
+        }
+      ]
+
+      return await prepareChangeQuote(walletSigner, txs, allocations)
+    },
+    fetchClaimQuote: (wallet: EdgeCurrencyWallet, requestAssetId: AssetId, nativeAmount: string) => {
+      throw new Error('fetchClaimQuote not implemented for TarotPoolAdapter')
+    },
+    fetchUnstakeExactQuote: async (wallet: EdgeCurrencyWallet, requestAssetId: AssetId, nativeAmount: string): Promise<ChangeQuote> => {
+      throw new Error('fetchUnstakeExactQuote not implemented for TarotPoolAdapter')
+    },
+    fetchStakePosition: async (wallet: EdgeCurrencyWallet): Promise<StakePosition> => {
+      const token0BorrowableContract = TarotBorrowable__factory.connect(token0BorrowableContractAddress, provider)
+      const token1BorrowableContract = TarotBorrowable__factory.connect(token1BorrowableContractAddress, provider)
+      const [token0, token1] = policyConfig.stakeAssets
+      const { walletAddress } = await workflowUtils(wallet)
+
+      const tarot = tarotUtils(adapterConfig, provider, walletAddress)
+
+      function getAllocation(token: AssetId, nativeAmount: string): PositionAllocation {
+        return {
+          pluginId: token.pluginId,
+          currencyCode: token.currencyCode,
+          allocationType: 'staked',
+          nativeAmount,
+          locktime: undefined
+        }
+      }
+
+      const allocations: PositionAllocation[] = []
+      const token0AccruedBalance = await tarot.getAccruedBalance(token0BorrowableContract, walletAddress)
+      const token0NativeAmount = token0AccruedBalance.toString()
+      allocations.push(getAllocation(token0, token0NativeAmount))
+
+      const token1AccruedBalance = await tarot.getAccruedBalance(token1BorrowableContract, walletAddress)
+      const token1NativeAmount = token1AccruedBalance.toString()
```

Comment:
I'd prefer naming `token1NativeAmount` to `token1AccruedBalance`. Ditch the other identifier by inlining the `toString()`: `(await tarot.getAccruedBalance(token1BorrowableContract, walletAddress)).toString`

Reply:
For debugging, I prefer to see exactly what the tarot utils blackbox produces before touching it. Not a hill I'm will to die on and can be convinced to change it

Reply:
If you want the extra variable for easy debugging later, then I'd at least prefer keeping the word "balance" in the final name. Usually I like identifier to tell me the semantic of the value and less about the type Ia "native amount" is a type, I'd argue). So I'm okay with `token1AccruedBalanceNativeAmount` or `token1AccruedBalanceString` (tack on the word `String` just so it doesn't conflict with the original identifier). Not a big deal this one, but I found it hard to know what token1NativeAmount meant (is it a balance, is it a parameter? What's the amount of?).